vulcan/lib/server/controllers/sessions_controller.rb

```diff
@@ -41,23 +42,44 @@ def orchestration_status(orchestration, run_errors)
         step = orchestration.workflow.find_step(step_name)
         next nil if step.nil?
         bt = orchestration.build_target_for(step_name, build_target_cache)
-        step_status_json(step.id, bt, run_errors)
+        step_status_json(
+          step_name: step.id,
+          bt: bt,
+          run_errors: run_errors,
+          ui_output: step.ui_output_name)
       end.select { |v| v }
     end
   end
 
-  def step_status_json(step_name, bt, run_errors=nil)
+  def step_status_json(step_name:, bt:, run_errors: nil, ui_output: false)
     {
       name: step_name,
-      status: run_errors&.include?(bt) ? 'error' : bt.is_built?(storage) ? 'complete' : 'pending',
+      status: run_errors&.include?(bt) ? 'error' : step_status(bt, ui_output),
       message: run_errors&.include?(bt) ? run_errors.message_for_build_target(bt) : nil,
-      downloads: bt.is_built?(storage) ? bt.build_outputs.map do |output_name, sf|
+      downloads: step_has_downloads?(bt, ui_output) ? bt.build_outputs.map do |output_name, sf|
         [
             output_name,
             storage.data_url(project_name: sf.project_name, cell_hash: sf.cell_hash, data_filename: sf.data_filename),
         ]
       end.to_h : nil
     }
   end
+
+  def step_status(bt, ui_output)
+    # UI Sinks (plot, download data, etc), will not have outputs
```

Comment:
Note that because UI output steps have no actual outputs, status has to be calculated differently.

Reply:
That is a good point.  Yeah, I'm ok with this.